### PR TITLE
Update deps and code style

### PR DESCRIPTION
**Pull request by gentoid**
I hope this project isn't abandoned. I wanted to add ability to handle mp3's too (and in the meantime, I'm not sure if I'm able to do that :-D), but at first I decided to update deps & code style

### CODE REVIEW

**Review by est31 — APPROVED** [review]
Thanks for the PR!

Yes, this library has been a bit under-maintained and since rodio arrived in the RustAudio org I'm not sure any more whether we should spend lots of resources on two different codebases that do the same thing. Especially as even more media frameworks like the one by rust-av are being built (as well as a certain unreleased one that one person loves to blog about).

The code changes look okay. I'm approving the PR but before merging will wait for what @mitchmindtree and @piedoom think about whether and how we should maintain audrey now that the more popular rodio crate is in our org.

**Comment by mitchmindtree** [comment]
@gentoid as mentioned in the nannou slack, thanks for this! I'll take a closer look tomorrow.

@est31 I always thought that rodio and audrey had quite significantly different goals - e.g. audrey is focused solely on abstracting over different kinds of audio format decoding (and hopefully encoding at some point), whereas rodio is an opinionated audio playback library built on top of CPAL originally designed to make it easier to play sounds and music in games. It's been a long time since I've looked closely at rodio, and things might have changed a lot since last I looked! I often personally still use audrey for my downstream projects on top of [nannou_audio](https://github.com/nannou-org/nannou_audio).

Perhaps the decoders should be removed from rodio in favour of using audrey internally to fill this role? If it turns out rodio's implementation of one of the supported formats is superior, perhaps it could be moved into audrey? This way we can reduce duplication and allow other kinds of audio applications to benefit from multi-format support?

In general I feel that rodio seems to conflate a lot of roles that could be abstracted into separate crates. E.g. its approach to sinks, sources and mixers could probably better abstracted in a more generalised DSP graph crate. It's reverberation and spatial effects could also be abstracted into their own crates for more general use. The Sample type abstractions and conversions could be more generally supported by a lower-level crate like `sample`. It's for these reasons I've never taken a great deal of interest in rodio - it seems to me more like tomaka's open source audio backend for simple games, whereas I personally am more interested in pro-audio, music playback, audio visual installations and performance, DAW development, etc and in turn am interested in more modular tools.

All that said, I can certainly understand why indie game devs would pick it up as the easiest way to get some audio into their game! I guess I'm just thinking we should find some way to consolidate it with the rest of the RustAudio ecosystem if we're going to take ownership of it and claim to continue to support it?

File: src/read.rs
Comment (mitchmindtree) [inline src/read.rs]: I've never seen this `r#` syntax in front of a `try!` macro invocation before! Is this a rustfmt bug I wonder? Either way, perhaps it's better to replace the use of the `try!` macro with the `?` operator these days.

File: src/read.rs
Comment (mitchmindtree) [inline src/read.rs]: And another - it looks like there's quite a few of these - it might be worth searching for al instances of `try!` and replacing them with `?` use. In the meantime I'll see if I can find the purpose of this `r#` syntax, I'm obviously missing something!

**Review by mitchmindtree — CHANGES_REQUESTED** [review]
Looking great in general, thanks again for this! I left a couple comments on some syntax I found odd, I'd be happy to merge this once those are addressed.

File: src/read.rs
Comment (est31) [reply src/read.rs]: `r#try!`is needed because `try` is a keyword in edition 2018. `r#` is needed to signify that it refers to an identifier and not the keyword. For more details [here](https://doc.rust-lang.org/edition-guide/rust-2018/module-system/raw-identifiers.html). The `try!` macro is being deprecated separately from that and I think nightly should already warn about the deprecation.

**Comment by gentoid** [comment]
woah, guys. You are awesome! You reviewed it so fast :+1:  I'll look carefully later today

**Comment by gentoid** [comment]
Reading the discussion. And I'd add my 2c. I'm personally much interested in music & DAW's. For now I'm working on an app, which later is going to be a music player, that passes my requirements. And who knows, maybe in the future I'll be so involved in this topic that I'll be able to work on a DAW app

File: src/read.rs
Comment (gentoid) [reply src/read.rs]: Yeah, I wasn't sure if I should use this syntax or `?`. So I just left as is after `cargo fix`. Will replace with `?` across all the changes

Comment (gentoid) [reply src/read.rs]: I've replaced `r#try!` with `?`

**Review by mitchmindtree — APPROVED** [review]

**Comment by mitchmindtree** [comment]
@gentoid all looks good to me, thanks again!

**Comment by gentoid** [comment]
Thanks guys! Together we did it really fast!

**Pull request merged**
